### PR TITLE
Cleanup Recycler to better encapsulate stuff

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We can move some methods etc to make encapsulation better in Recycler

Modifications:

Move / rename methods to make usage more clear

Result:

Code cleanup

### CODE REVIEW

**Comment by normanmaurer** [comment]
@njhill just some cleanup on top of yours

File: common/src/main/java/io/netty/util/Recycler.java [hunk 1]
```diff
@@ -665,6 +669,13 @@ private void pushLater(DefaultHandle<?> item, Thread thread) {
             queue.add(item);
         }
 
+        /**
+         * Allocate a new {@link WeakOrderQueue} or return {@code null} if not possible.
+         */
+        WeakOrderQueue newWeakOrderQueue(Thread thread) {
```
Comment (njhill) [inline common/src/main/java/io/netty/util/Recycler.java]: private?

File: common/src/main/java/io/netty/util/Recycler.java [hunk 2]
```diff
@@ -272,28 +272,37 @@ void reclaimAllSpaceAndUnlink() {
...
-            void reclaimSpace(int space) {
-                assert space >= 0;
+            private void reclaimSpace(int space) {
                 availableSharedCapacity.addAndGet(space);
             }
 
-            boolean reserveSpace(int space) {
-                return reserveSpace(availableSharedCapacity, space);
+            void relink(Link link) {
+                if (this.link != link) {
```
Comment (njhill) [inline common/src/main/java/io/netty/util/Recycler.java]: I don't think this check was there before, was it intentional?

If so then the `this.link = link;` below could be moved inside the if block?

Comment (normanmaurer) [reply common/src/main/java/io/netty/util/Recycler.java]: I think we can just remove it...

**Comment by normanmaurer** [comment]
@njhill PTAL again

**Comment by normanmaurer** [comment]
@franz1981 ping

File: common/src/main/java/io/netty/util/Recycler.java [hunk 3]
```diff
@@ -272,28 +272,35 @@ void reclaimAllSpaceAndUnlink() {
...
 
-            void reclaimSpace(int space) {
-                assert space >= 0;
+            private void reclaimSpace(int space) {
                 availableSharedCapacity.addAndGet(space);
             }
 
-            boolean reserveSpace(int space) {
-                return reserveSpace(availableSharedCapacity, space);
+            void relink(Link link) {
```
Comment (franz1981) [inline common/src/main/java/io/netty/util/Recycler.java]: final?

**Pull request merged**